cpp/src/arrow/csv/reader_test.cc

```diff
@@ -358,6 +358,44 @@ TEST(StreamingReaderTest, BytesRead) {
   }
 }
 
+TEST(StreamingReaderTests, SkipMultipleEmptyBlocksAtStart) {
+  ASSERT_OK_AND_ASSIGN(auto thread_pool, internal::ThreadPool::Make(1));
+  auto table_buffer = std::make_shared<Buffer>(
+      "aaa,bbb,ccc\n123,456,789\n101,112,131\n415,161,718\n192,021,222\n324,252,627\n"
+      "282,930,313\n233,343,536\n");
+
+  auto input = std::make_shared<io::BufferReader>(table_buffer);
+
+  auto read_options = ReadOptions::Defaults();
+  read_options.block_size = 34;
+  read_options.skip_rows_after_names = 6;
+
+  ASSERT_OK_AND_ASSIGN(
+      auto streaming_reader,
+      StreamingReader::Make(io::default_io_context(), input, read_options,
+                            ParseOptions::Defaults(), ConvertOptions::Defaults()));
+  std::shared_ptr<RecordBatch> batch;
+  ASSERT_EQ(12, streaming_reader->bytes_read());
+
+  // The first batch should have the one and only row in it
+  ASSERT_OK(streaming_reader->ReadNext(&batch));
+  ASSERT_NE(nullptr, batch.get());
+  ASSERT_EQ(1, batch->num_rows());
+  ASSERT_EQ(96, streaming_reader->bytes_read());
+
+  auto schema = streaming_reader->schema();
+  ASSERT_EQ(3, schema->num_fields());
+  ASSERT_EQ("aaa", schema->field(0)->name());
+  ASSERT_EQ(Type::INT64, schema->field(0)->type()->id());
+  ASSERT_EQ("bbb", schema->field(1)->name());
+  ASSERT_EQ(Type::INT64, schema->field(1)->type()->id());
+  ASSERT_EQ("ccc", schema->field(2)->name());
+  ASSERT_EQ(Type::INT64, schema->field(2)->type()->id());
+
```

Comment:
Can you also test the contents of `batch`? You can use `RecordBatchFromJSON` to construct the expected record batch.